pkg/controller/job/job_controller.go

```diff
@@ -973,24 +974,27 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 // This is done through DELETE calls that set deletion timestamps.
 // The method trackJobStatusAndRemoveFinalizers removes the finalizers, after
 // which the objects can actually be deleted.
-// Returns number of successfully deletions issued.
-func (jm *Controller) deleteActivePods(ctx context.Context, job *batch.Job, pods []*v1.Pod) (int32, error) {
+// Returns number of successfully deleted ready pods and total number of successfully deleted pods.
+func (jm *Controller) deleteActivePods(ctx context.Context, job *batch.Job, pods []*v1.Pod) (deletedReadyPods, deletedPods int32, err error) {
 	errCh := make(chan error, len(pods))
-	successfulDeletes := int32(len(pods))
+	deletedPods = int32(len(pods))
```

Comment:
I think your name is ok, but I would suggest to minimize the diff to changes which are essential for the fix.